### PR TITLE
[WIP] Return a warning when the state-into-spec annotation will be defaulted to 'merge'

### PR DESCRIPTION
**Pull request by maqiuyujoyce**
### Change description

This is an experiment PR that checks the default value of `state-into-spec` annotation in the webhook via retrieving the ConfigConnector object and potentially the ConfigConnectorContext object, and returns a warning if the default value is `merge`.

In general, we shouldn't make API calls in the webhook because it has a short timeout (10s), though it should be fairly cheap to do the CC/CCC object retrievals.

### Tests you have done

- [N/A] Run `make ready-pr` to ensure this PR is ready for review.
- [X] Perform necessary E2E testing for changed resources.

Ran `make deploy-controller` locally with basic CCC and CC objects (no `spec.stateIntoSpec` configuration), then applied a PubsubTopic resource (no `state-into-spec` annotation). Got the following warning as expected:

```
kubectl apply -f pubsubtopic.yaml 
Warning: 'cnrm.cloud.google.com/state-into-spec: merge' is unsupported for CRDs added in 1.114.0 and later. Use 'absent' instead. More details can be found at https://cloud.google.com/config-connector/docs/concepts/ignore-unspecified-fields.
pubsubtopic.pubsub.cnrm.cloud.google.com/pubsubtopic-sample-2 created
```

### CODE REVIEW

File: pkg/webhook/state_into_spec_annotation_validator.go [hunk 1]
```diff
@@ -51,13 +51,28 @@ func (a *stateIntoSpecAnnotationValidator) Handle(ctx context.Context, req admis
 	}
 
 	value, ok := k8s.GetAnnotation(k8s.StateIntoSpecAnnotation, obj)
+	klog.Infof("maqiuyu...annotation value %v, ok? %v", value, ok)
```
Comment (gemmahou) [inline pkg/webhook/state_into_spec_annotation_validator.go:54]: nit: remove the debug message?

Comment (maqiuyujoyce) [reply pkg/webhook/state_into_spec_annotation_validator.go]: Thanks, @gemmahou ! Will clean up if we decide we'll move forward with this change.

**Comment by maqiuyujoyce** [comment]
I think it's nice to have the warning in the webhook, though making API calls in the webhook seems to be violating the best practices and may have consequence about the performance and etc. IMO the risk is higher than the benefit, and it should be okay if we don't merge this PR (e.g. users don't receive warnings if their CRs are defaulted with `state-into-spec: merge`). @yuwenma WDYT?

File: pkg/webhook/state_into_spec_annotation_validator.go [hunk 2]
```diff
@@ -51,13 +51,29 @@ func (a *stateIntoSpecAnnotationValidator) Handle(ctx context.Context, req admis
 	}
 
 	value, ok := k8s.GetAnnotation(k8s.StateIntoSpecAnnotation, obj)
+	klog.Infof("maqiuyu...kind %q, obj %q, annotation value %v, ok? %v", obj.GroupVersionKind().Kind, obj.GetName(), value, ok)
```
Comment (yuwenma) [inline pkg/webhook/state_into_spec_annotation_validator.go:54]: nit: clean up?

**Comment by maqiuyujoyce** [comment]
Discussed offline and decided not to add this feature.

**Pull request closed without merging**